packages/cursorless-engine/src/processTargets/modifiers/RelativeScopeStage.ts

```diff
@@ -130,12 +140,79 @@ function generateScopesExclusive(
     ? "disallowed"
     : "disallowedIfStrict";
 
-  return islice(
+  let scopes = scopeHandler.generateScopes(editor, initialPosition, direction, {
+    containment,
+    skipAncestorScopes: true,
+  });
+
+  const interiorRanges = getInteriorRanges(
+    scopeHandlerFactory,
+    scopeHandler,
+    editor,
+    initialPosition,
+    direction,
+  );
+
+  if (interiorRanges != null) {
+    scopes = ifilter(
+      scopes,
+      (s) => !interiorRanges.some((r) => r.contains(s.domain)),
+    );
+  }
+
+  return islice(scopes, offset - 1, offset + desiredScopeCount - 1);
+}
+
+function getInteriorRanges(
+  scopeHandlerFactory: ScopeHandlerFactory,
+  scopeHandler: ScopeHandler,
+  editor: TextEditor,
+  initialPosition: Position,
+  direction: Direction,
+): Range[] | undefined {
+  const interiorScopeHandler = scopeHandlerFactory.maybeCreate(
+    { type: "interior" },
+    editor.document.languageId,
+  );
+
+  if (interiorScopeHandler == null) {
+    return undefined;
+  }
+
+  const containingScope = find(
     scopeHandler.generateScopes(editor, initialPosition, direction, {
-      containment,
+      containment: "required",
+      allowAdjacentScopes: true,
       skipAncestorScopes: true,
     }),
-    offset - 1,
-    offset + desiredScopeCount - 1,
   );
+
+  if (containingScope == null) {
+    return undefined;
+  }
+
+  const containingInitialPosition =
+    direction === "forward"
+      ? containingScope.domain.start
+      : containingScope.domain.end;
+  const containingDistalPosition =
+    direction === "forward"
+      ? containingScope.domain.end
+      : containingScope.domain.start;
+
+  const interiorScopes = interiorScopeHandler.generateScopes(
+    editor,
+    containingInitialPosition,
+    direction,
+    {
+      skipAncestorScopes: true,
+      distalPosition: containingDistalPosition,
+    },
+  );
+
+  // Interiors containing the initial position are excluded
+  const interiorRanges = Array.from(interiorScopes)
+    .filter((s) => !s.domain.contains(initialPosition))
+    .map((s) => s.domain);
```

Comment:
Nitpicking here but might as well do map / filter on the stream then convert to list at the end